src/pybind/mgr/cephadm/agent.py

```diff
@@ -54,7 +55,18 @@ def __init__(self, mgr: "CephadmOrchestrator") -> None:
 
     def run(self) -> None:
         try:
-            self.ssl_certs.generate_root_cert()
+            try:
+                old_creds = self.mgr.get_store('cephadm_endpoint_credentials')
+                assert old_creds
+                old_creds_dict = json.loads(old_creds)
+                old_key = old_creds_dict['key']
+                old_cert = old_creds_dict['cert']
+                self.ssl_certs.load_root_credentials(old_cert, old_key)
+            except AssertionError:
```

Comment:
`except AssertionError:` in production code is bad practice.

Reply:
Removed this. For future reference, why is this considered bad practice?

Reply:
There are some good use cases for asserts:

* verify things in unit tests (pytest)
* tell mypy a type is reduced at this point (e.g. we know `Optional[str]` is never `None` here)
* Asserts should be used to test conditions that should never happen. The purpose is to crash early in the case of a corrupt program state.
* -O and -OO python arguemnts will strip away your assert statements